org.eclipse.lsp4j/src/main/java/org/eclipse/lsp4j/Protocol.xtend

```diff
@@ -3910,13 +3911,13 @@ class ProgressParams {
 	 * The progress data.
 	 */
 	@NonNull
-	@JsonAdapter(WorkDoneProgressNotificationAdapter.Factory)
-	WorkDoneProgressNotification value
+	@JsonAdapter(ProgressNotificationAdapter)
+	Either<? extends WorkDoneProgressNotification, ? extends Object> value
```

Comment:
why `? extends` - this is also inconsistent with for e.g. `SemanticTokensClientCapabilitiesRequests.range`.

Reply:
Without `? extends`, it did not compile:

```
Cannot cast from Procedures.Procedure1<JsonObject> to Procedures.Procedure1<? super Object>	JsonParseTest.java	/org.eclipse.lsp4j/src/test/xtend-gen/org/eclipse/lsp4j/test/services	line 3461	Java Problem
Cannot cast from Procedures.Procedure1<JsonObject> to Procedures.Procedure1<? super Object>	JsonSerializeTest.java	/org.eclipse.lsp4j/src/test/xtend-gen/org/eclipse/lsp4j/test/services	line 1876	Java Problem
Java problem: Cannot cast from Procedures.Procedure1<JsonObject> to Procedures.Procedure1<? super Object>	JsonParseTest.xtend	/org.eclipse.lsp4j/src/test/java/org/eclipse/lsp4j/test/services	line 1613	Xtend Problem
Java problem: Cannot cast from Procedures.Procedure1<JsonObject> to Procedures.Procedure1<? super Object>	JsonSerializeTest.xtend	/org.eclipse.lsp4j/src/test/java/org/eclipse/lsp4j/test/services	line 910	Xtend Problem
```

Probably this is an Xtend code generation issue? Not sure how to fix it.

Reply:
👍 I'll have a look before merging to see what is going on here.

Reply:
Thanks Jonah!

Reply:
The solution to removing the `? extends` is like this in the .xtend file:

```java
			params = new ProgressParams => [
				token = Either.forRight(1234)
				value = Either.forRight((new JsonObject => [
					addProperty("foo", "bar")
				]) as Object)
			]
```

i.e. upcast the `JsonObject` as `Object`. 

or specify upfront which types in the Either generics we want:

```java
			params = new ProgressParams => [
				token = Either.forRight(1234)
				value = Either.<WorkDoneProgressNotification, Object>forRight(new JsonObject => [
					addProperty("foo", "bar")
				])
			]
```

I do think there is an argument to be made that all the non-final types should be `? extends` in Protocol, but I think for now consistency is better and if you think it should be `? extends` then we should handle that in a new issue.

WDYT? If you like one of the above solutions, can you update the PR.

Reply:
Done!

> I do think there is an argument to be made that all the non-final types should be ? extends in Protocol, but I think for now consistency is better and if you think it should be ? extends then we should handle that in a new issue.

I fully agree.